infrastructure/crypto/src/ristretto/ristretto_keys.rs

```diff
@@ -489,4 +490,18 @@ mod test {
         let (k, pk) = RistrettoPublicKey::random_keypair(&mut rng);
         assert_eq!(pk, RistrettoPublicKey::from_secret_key(&k));
     }
+
+    #[test]
+    fn secret_keys_are_cleared_after_drop() {
+        let zero = &vec![0u8; 32][..];
+        let mut rng = rand::OsRng::new().unwrap();
+        let ptr;
+        {
+            let k = RistrettoSecretKey::random(&mut rng);
+            ptr = (k.0).as_bytes().as_ptr();
+        }
+        unsafe {
+            assert_eq!(slice::from_raw_parts(ptr, 32), zero);
```

Comment:
I might be wrong in this, but this checks that the memory contains 0's. 
But what if something else was written here in the meantime between cleaning it and reading again?
Would it not be better to check that it is not k?

Reply:
You're right, it checks that memory is zeroed, because that's what my expectation is.
If not, the test should fail (as it has been until the last commit)

I'd be very worried if something was writing to that memory space between line 502 and 504. Nothing should be able to write to that address space. This would indicate a very broken OS security model.